segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AbstractConnectionListener.java

```diff
@@ -224,4 +271,37 @@ public void close() {
             tlsCertFileModificationMonitor.stopMonitoring();
         }
     }
+
+    public static class ConnectionListenerHealthContributor extends AbstractHealthContributor {
```

Comment:
Does this class need to be public? What is the minimum possible visibility it may have which still allows it to serve its purpose?

Same question applies to its constructor below (and in general to any method/class that you have - give them the minimum needed visibility).